docs/WindowsBuild.md

```diff
@@ -0,0 +1,176 @@
+pushd "%swift_source_dir%/build/Ninja-RelWithDebInfoAssert/llvm-windows-amd64"
+cmake -G "Ninja"^
+ -DLLVM_ENABLE_ASSERTIONS=TRUE^
+ -DCMAKE_BUILD_TYPE=RelWithDebInfo^
+ -DLLVM_TOOL_SWIFT_BUILD=NO^
+ -DLLVM_INCLUDE_DOCS=TRUE^
+ -DLLVM_TOOL_COMPILER_RT_BUILD=TRUE^
+ -DLLVM_BUILD_EXTERNAL_COMPILER_RT=TRUE^
+ -DLLVM_LIT_ARGS=-sv^
+ -DLLVM_TARGETS_TO_BUILD=X86^
+ "%swift_source_dir%/llvm"
+popd
+cmake --build "%swift_source_dir%/build/Ninja-DebugAssert/llvm-windows-amd64"
+```
+
+### 7. Build Swift
+- This must be done from within a developer command prompt and could take up to
+  two hours depending on your system.
+- You may need to adjust the `SWIFT_WINDOWS_LIB_DIRECTORY` parameter depending on
+  your target platform or Windows SDK version.
+- While the commands here use MSVC to build, using clang-cl is recommended (see
+  the **Clang-cl** section below).
+```cmd
+mkdir "%swift_source_dir%/build/Ninja-DebugAssert/swift-windows-amd64/ninja"
+pushd "%swift_source_dir%/build/Ninja-DebugAssert/swift-windows-amd64/ninja"
+cmake -G "Ninja" "%swift_source_dir%/swift"^
+ -DCMAKE_BUILD_TYPE=Debug^
+ -DSWIFT_PATH_TO_CMARK_SOURCE="%swift_source_dir%/cmark"^
+ -DSWIFT_PATH_TO_CMARK_BUILD="%swift_source_dir%/build/Ninja-DebugAssert/cmark-windows-amd64"^
+ -DSWIFT_CMARK_LIBRARY_DIR="%swift_source_dir%/build/Ninja-DebugAssert/cmark-windows-amd64/src"^
+ -DSWIFT_PATH_TO_LLVM_SOURCE="%swift_source_dir%/llvm"^
+ -DSWIFT_PATH_TO_LLVM_BUILD="%swift_source_dir%/build/Ninja-RelWithDebInfoAssert/llvm-windows-amd64"^
+ -DSWIFT_PATH_TO_CLANG_SOURCE="%swift_source_dir%/llvm/tools/clang"^
+ -DSWIFT_PATH_TO_CLANG_BUILD="%swift_source_dir%/build/Ninja-RelWithDebInfoAssert/llvm-windows-amd64"^
+ -DICU_UC_INCLUDE_DIRS="%swift_source_dir%/icu/include"^
+ -DICU_UC_LIBRARY_DIRS="%swift_source_dir%/icu/lib64"^
+ -DICU_I18N_INCLUDE_DIRS="%swift_source_dir%/icu/include"^
+ -DICU_I18N_LIBRARY_DIRS="%swift_source_dir%/icu/lib64"^
+ -DICU_UC_LIB_NAME="icuuc"^
+ -DICU_I18N_LIB_NAME="icuin"^
+ -DSWIFT_INCLUDE_DOCS=FALSE^
+ -DSWIFT_INCLUDE_TESTS=FALSE^
+ -DSWIFT_BUILD_DYNAMIC_SDK_OVERLAY=FALSE^
+ -DSWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER=FALSE
+popd
+cmake --build "%swift_source_dir%/build/Ninja-DebugAssert/swift-windows-amd64/ninja"
+```
+
+- To create a Visual Studio project, you'll need to change the generator and,
+  if you have a 64 bit processor, specify the generator platform. Note that you
+  may get multiple build errors compiling the `swift` project due to an MSBuild
+  limitation that file paths cannot exceed 260 characters. These can be
+  ignored, as they occur after the build when writing the last build status to
+  a file.
+```cmd
+cmake -G "Visual Studio 15" "%swift_source_dir%/swift"^
+ -DCMAKE_GENERATOR_PLATFORM="x64"^
+ ...
+```
+
+## Clang-cl
+
+Follow the instructions for MSVC, but add the following lines to each CMake
+configuration command. `Clang-cl` 4.0.1 has been tested. You can remove the
```

Comment:
Nit: no need for code voice here.

Reply:
Why do you think so? My intuition says yes. We're referring to the binary version of the tool, not the project itself.